api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -641,6 +650,13 @@ public int hashCode() {
       }
       return fieldsById;
     }
+
+    /**
+     * @return true if struct represents union schema
+     */
+    public boolean isUnionSchema() {
```

Comment:
Why do we need this in the main API?  The API should not expose unions or info about union. Can't we do the translation in during data reading?
Both Avro and ORC will take read schemas. IMO if the underlying file schema is a union and the read schema is a struct, the translation should happen there.

Reply:
I will check if we can make it package private, Avro visitor needs it so that it can add property in schema to reflect union type.

Reply:
I agree. Iceberg's type system should have no knowledge of union types because unions aren't supported.

Reply:
I think this is naming issue. this flag indicate that , field is storing union type by converting it to record/struct.